tornado_jsonapi/handlers.py

```diff
@@ -17,194 +17,248 @@
-            ensure_ascii=False, indent=4))
+
+        if additional:
+            data.update(additional)
+        data.update(dict(data=json_resources, **self._get_meta()))
+
+        self.finish(json.dumps(data, ensure_ascii=False, indent=4))
 
     def _get_request_data(self, schema):
-        '''
+        """
         Get and validate request JSON data according to given schema
-        '''
+        """
         try:
-            d = json.loads(self.request.body.decode(encoding='UTF-8'))
+            d = json.loads(self.request.body.decode(encoding="UTF-8"))
         except ValueError as err:
-            raise APIError(
-                status.HTTP_400_BAD_REQUEST,
-                str(err)) from err
+            raise APIError(status.HTTP_400_BAD_REQUEST, str(err)) from err
         try:
             data = schema(**d)
             data.validate()
         except pjs.validators.ValidationError as err:
-            raise APIError(
-                status.HTTP_400_BAD_REQUEST,
-                str(err)) from err
-        return data['data']
+            raise APIError(status.HTTP_400_BAD_REQUEST, str(err)) from err
+        return data["data"]
 
     def _get_resource(self, data, validate=True):
-        if data.get('type') != self._resource.name():
+        if data.get("type") != self._resource.name():
             raise APIError(
                 status.HTTP_409_CONFLICT,
-                'Expecting object of type "%s"', self._resource.name())
-        if data['attributes'] is None:
-            raise APIError(
-                status.HTTP_400_BAD_REQUEST,
-                'Missing object attributes')
-        attributes = data['attributes'].as_dict()
+                'Expecting object of type "%s"',
+                self._resource.name(),
+            )
+        if data["attributes"] is None:
+            raise APIError(status.HTTP_400_BAD_REQUEST, "Missing object attributes")
+        attributes = data["attributes"].as_dict()
         if validate:
             try:
                 attrs = self._resource._schema(**attributes)
                 attrs.validate()
             except pjs.validators.ValidationError as err:
-                raise APIError(
-                    status.HTTP_400_BAD_REQUEST,
-                    str(err)) from err
+                raise APIError(status.HTTP_400_BAD_REQUEST, str(err)) from err
         return attributes
 
     @tornado.gen.coroutine
     def get(self, id_=None):
-        '''
+        """
         GET method, see
         `spec <http://jsonapi.org/format/1.0/#fetching-resources>`__.
         Decorate with :py:func:`tornado.gen.coroutine` when subclassing.
-        '''
+        """
+
+        limit = (
+            int(self.request.arguments["limit"][0])
+            if "limit" in self.request.arguments
+            else 0
+        )
+        server_limit = (
+            self.settings["jsonapi_limit"] if "jsonapi_limit" in self.settings else 0
+        )
+        print("server limit", server_limit)
```

Comment:
I suggest removing any debug `print`ing or using `logging` for that purpose (if you really need it).